From ec3e2631ae789582da68df023074c0844a1c6985 Mon Sep 17 00:00:00 2001
From: Charles Bousseau <cbousseau@anaconda.com>
Date: Fri, 30 May 2025 15:38:05 -0400
Subject: [PATCH] make pydeck optional

---
 lib/setup.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/lib/setup.py b/lib/setup.py
index f600acfc7..f5e9cfbbc 100644
--- a/lib/setup.py
+++ b/lib/setup.py
@@ -62,7 +62,6 @@ INSTALL_REQUIRES = [
 # `pip install streamlit` or `conda install -c conda-forge streamlit`)
 SNOWPARK_CONDA_EXCLUDED_DEPENDENCIES = [
     "gitpython>=3.0.7, <4, !=3.1.19",
-    "pydeck>=0.8.0b4, <1",
     # Tornado 6.0.3 was the current version when Python 3.8 was released (Oct 14, 2019).
     "tornado>=6.0.3, <7",
 ]
@@ -74,11 +73,17 @@ EXTRA_REQUIRES = {
     "snowflake": [
         "snowflake-snowpark-python[modin]>=1.17.0; python_version<'3.12'",
         "snowflake-connector-python>=3.3.0; python_version<'3.12'",
+    ],
+    # pydeck 0.9.1 and below restrict ipywidgets to version 7
+    # This leads to non-functional systems when installed with recent jupyterlab versions.
+    # Pydeck supports creating interactive map visualization, which is not the main feature of streamlit.
+    "pydeck": [
+        "pydeck>=0.8.0b4, <1",
     ]
 }
 
 
-class VerifyVersionCommand(install):
+class VerifyVersionCommand(install):#
     """Custom command to verify that the git tag matches our version."""
 
     description = "verify that the git tag matches our version"
-- 
2.39.5 (Apple Git-154)

