From 08a66437d5115f7254043ea8adfaf8e2d9d7fc7a Mon Sep 17 00:00:00 2001
From: Patryk Kups <pkups@anaconda.com>
Date: Thu, 5 Feb 2026 09:34:15 +0100
Subject: [PATCH] move pydeck to extra deps

---
 pyproject.toml | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 01eac4e..eaa052d 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -69,7 +69,6 @@ dependencies = [
   # the index dtype.
   "pandas>=1.4.0,<3",
   "pillow>=7.1.0,<13",
-  "pydeck>=0.8.0b4,<1",
   # `protoc` < 3.20 is not able to generate protobuf code compatible with protobuf >= 3.20.
   "protobuf>=3.20,<7",
   # pyarrow is not semantically versioned, gets new major versions frequently, and
@@ -117,6 +116,10 @@ starlette = [
 pdf = ["streamlit-pdf>=1.0.0"]
 # Optional dependency required for auth:
 auth = ["Authlib>=1.3.2"]
+# pydeck 0.9.1 and below restrict ipywidgets to version 7
+# This leads to non-functional systems when installed with recent jupyterlab versions.
+# Pydeck supports creating interactive map visualization, which is not the main feature of streamlit.
+visualization = ["pydeck>=0.8.0b4, <1"]
 # Optional charting dependencies:
 charts = [
   "matplotlib>=3.0.0",
@@ -138,7 +141,7 @@ performance = [
 ]
 # Install all optional dependencies:
 all = [
-  "streamlit[auth,charts,snowflake,sql,pdf,performance]",
+  "streamlit[auth,visualization,charts,snowflake,sql,pdf,performance]",
   # Improved exception traceback formatting:
   "rich>=11.0.0",
 ]
-- 
2.39.5 (Apple Git-154)

