From 2d8012b60da1b7ed2d0ee1a99d06a85fdc1d7569 Mon Sep 17 00:00:00 2001
From: Andrii Osipov <aosipov@anaconda.com>
Date: Mon, 10 Feb 2025 15:32:10 -0800
Subject: [PATCH] fix win encoding

---
 lib/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/setup.py b/lib/setup.py
index 81d92c4..9d16849 100644
--- a/lib/setup.py
+++ b/lib/setup.py
@@ -92,7 +92,7 @@ class VerifyVersionCommand(install):
 
 readme_path = THIS_DIRECTORY / ".." / "README.md"
 if readme_path.exists():
-    long_description = readme_path.read_text()
+    long_description = readme_path.read_text(encoding='utf-8')
 else:
     # In some build environments (specifically in conda), we may not have the README file
     # readily available. In these cases, just let long_description be the empty string.
-- 
2.39.3 (Apple Git-146)

